include/windows/osd.h

```diff
@@ -724,6 +734,20 @@ static inline ssize_t ofi_send_socket(SOCKET fd, const void *buf, size_t count,
 	return send(fd, (const char*)buf, (int)count, flags);
 }
 
+ssize_t sendmsg(SOCKET sd, const struct msghdr *msg, int flags);
+
+static inline ssize_t ofi_sendmsg_socket(SOCKET fd, const struct msghdr *msg, int flags)
+{
+	return sendmsg(fd, msg, flags);
```

Comment:
There is this unfortunate limitation on windows platforms for sendto (which is used in sendmsg implementation in osd.c). Unix/Linux it is okay. 

from https://docs.microsoft.com/en-us/windows/desktop/api/winsock/nf-winsock-sendto
> The sendto function is used to write outgoing data on a socket. For message-oriented sockets, care must be taken not to exceed the maximum packet size of the underlying subnets, which can be obtained by using getsockopt to retrieve the value of socket option SO_MAX_MSG_SIZE. If the data is too long to pass atomically through the underlying protocol, the error WSAEMSGSIZE is returned and no data is transmitted

>

Reply:
Thanks @vkrishna, that's a good catch. I wasn't aware of this limitation. 
I should probably not use `sendmsg` for connection-oriented sockets. 
I'll think about an alternative solution.

Reply:
Doesn't the end up picking up the windows osd version of the call, rather than the MS version?

Reply:
The implementation of sendmsg in osd.c uses MS' sendto call.

Reply:
@vkrishna @shefty I've added a new patch in the series ('src/windows/osd: modify sendmsg function for connected sockets'), which should workaround the issue with senmsg() and unconnected sockets on Windows. Please let me know if you agree with this change.

Reply:
I didn't catch this before, but the reference to sendto() doesn't apply here.  The limit is only for message-oriented sockets, which does not apply for stream-oriented sockets.  So, we don't need to distinguish between using send versus sendto. The bigger issue is that the windows abstractions are allocating and copying the buffers being transferred.

Reply:
oh I see, message oriented means UDP. Sorry for confusion.